tcms/core/ajax.py

```diff
@@ -137,23 +136,23 @@ def tags(self):
         def users(self):
             from django.contrib.auth.models import User
 
-            query = strip_parameters(self.request, self.internal_parameters)
+            query = strip_parameters(self.request.GET, self.internal_parameters)
             return User.objects.filter(**query)
 
         def versions(self):
             from tcms.management.models import Version
 
             return Version.objects.filter(product__id=self.product_id)
 
-    objects = Objects(request=request, product_id=request.GET['product_id'])
+    objects = Objects(request=request, product_id=request.GET.get('product_id'))
     obj = getattr(objects, request.GET['info_type'], None)
 
     if obj:
         if request.GET.get('format') == 'ulli':
             field = request.GET.get('field', 'name')
             response_str = '<ul>'
-            for o in obj():
-                response_str += '<li>' + getattr(o, field, None) + '</li>'
+            for o in obj().values(field):
+                response_str += '<li>' + o.get(field, None) + '</li>'
```

Comment:
`obj()` is the `tags()` method from `ajax.py` which returns a query set and then we loop over the objects returned from this query set. Since we only need the value of a single field in this query set `values(field)` makes a small performance optimization and returns the values only for that single field. `obj().values()` in this case is the `QuerySet.values` method.